### PR TITLE
feat: Detect that the completePartitionStart() fails due to shutdown …

### LINKED ISSUE #14510: Cancelled partition startup should not be logged as error
When the partition startup processed is cancelled because shutdown was called before startup finished, the `PartitionManager` logs this as error:

```
Failed to start partition 2: io.camunda.zeebe.scheduler.startup.StartupProcessException: Aborting startup process because shutdown was called
	at io.camunda.zeebe.scheduler.startup.StartupProcess.proceedWithStartupSynchronized(StartupProcess.java:153) ~[zeebe-scheduler-8.3.0-SNAPSHOT.jar:8.3.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.startup.StartupProcess.lambda$proceedWithStartupSynchronized$3(StartupProcess.java:169) ~[zeebe-scheduler-8.3.0-SNAPSHOT.jar:8.3.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:28) [zeebe-scheduler-8.3.0-SNAPSHOT.jar:8.3.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:94) [zeebe-scheduler-8.3.0-SNAPSHOT.jar:8.3.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:45) [zeebe-scheduler-8.3.0-SNAPSHOT.jar:8.3.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:119) [zeebe-scheduler-8.3.0-SNAPSHOT.jar:8.3.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:130) [zeebe-scheduler-8.3.0-SNAPSHOT.jar:8.3.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:108) [zeebe-scheduler-8.3.0-SNAPSHOT.jar:8.3.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:227) [zeebe-scheduler-8.3.0-SNAPSHOT.jar:8.3.0-SNAPSHOT]
```

This can happen when a broker is shut down shortly after being started or when a partition has never finished the startup process because it is stuck trying to join the raft partition.

We should detect this case and not log it as error.

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

Detect that the completePartitionStart() fails due to shutdown being called and log a warning instead.

## Related issues

closes #14510 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerImpl.java
Comment (deepthidevaki) [inline broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerImpl.java:188]: :x: 
```suggestion
      if (error instanceof StartupProcessShutdownException) {
```

File: broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerImpl.java
Comment (deepthidevaki) [inline broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerImpl.java:189]: :wrench: I think debug log is enough. 
```suggestion
        LOGGER.debug("Aborting startup of partition {}", partitionId, error);
```

**Review by deepthidevaki — APPROVED** [review]
:100: Fix looks good. Please address my comments before merging. No need for another review.

**Comment by rodrigo-lourenco-lopes** [comment]
bors merge

**Pull request merged**
